### PR TITLE
[FSPE-5350] Implemented Sidenav Open-on-hover behavior

### PR DESCRIPTION
**Pull request by janibolkvadze**
- [x] Added optional `open-on-hover` behavior.
- [x] Changed Drawer behavior by adding `persist-on-click` behavior (Once Sidenav Drawer is open and the user clicks Selected menu item again, it does not close anymore)
- [x] Changed the way Sidenav helper class names are defined
- [x] Added `Open on hover` example to documentation

@dani-cl-madrid @SergioQCL Once we approve the changes, I'll format the file, since there are many formatting errors to fix.

https://ctl.atlassian.net/browse/FSPE-5350

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://0786bff68ac6f960256177112e5aeed87c749f50-chi-pr.enterprise.ctl.io" target="_blank"> https://0786bff68ac6f960256177112e5aeed87c749f50-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/1124/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1125/ for more details.

**Comment by mattnickles** [comment]
@janibolkvadze To meet the request we'll need to display the drawer on hover in both the open and closed state. This might be a bug but currently menu item 1 is not applying `-active` to `drawer-1-1` on hover (in a closed state) until the drawer has been opened and closed via click. To reproduce, click menu item 1, `drawer-1-1` will launch and get set to `-active`, click close button, hover over menu item 1, and `drawer-1-1` will now launch and get set to `-active` via hover.

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/1126/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://0e1477e39a628a683183c618d88dd7d1e3933567-chi-pr.enterprise.ctl.io" target="_blank"> https://0e1477e39a628a683183c618d88dd7d1e3933567-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1127/ for more details.

File: src/chi/javascript/components/sidenav.js [hunk 1]
```diff
@@ -98,6 +102,44 @@ class Sidenav extends Component {
...
+    const enableAnimation = () => {
+      this._drawers.forEach((otherDrawer) => {
+        otherDrawer._config.animated = true;
+      });
+    };
+    const _openDrawerOnMouseEnter = (drawer) => {
+      drawer.show();
+      disableAnimation(drawer);
+    };
+    const _closeDrawerOnMouseleave = (drawer) => {
```
Comment (mattnickles) [inline src/chi/javascript/components/sidenav.js:130]: @janibolkvadze Hover behavior looks great. Click behavior is a little buggy - the drawer abruptly closes when any first tier menu item is clicked. Is the drawer not animating on close due to -active getting removed before -transitioning has completed?

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://81abca8be507eb92b02e6ffac68ab96178d90821-chi-pr.enterprise.ctl.io" target="_blank"> https://81abca8be507eb92b02e6ffac68ab96178d90821-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1137/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://a8fc2f87e3cf0b188c5ce7bade5e81f987ccfb45-chi-pr.enterprise.ctl.io" target="_blank"> https://a8fc2f87e3cf0b188c5ce7bade5e81f987ccfb45-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1138/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://90b663fc3b1610b35b21e3aa9a34aff970505fd2-chi-pr.enterprise.ctl.io" target="_blank"> https://90b663fc3b1610b35b21e3aa9a34aff970505fd2-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1141/ for more details.

**Comment by jllr** [comment]
@dani-cl-madrid and @SergioQCL can you review this PR? Thanks

**Comment by janibolkvadze** [comment]
> @dani-cl-madrid and @SergioQCL can you review this PR? Thanks

I sent this commit because I needed to generate web instance to test the behavior on Tablet / Mobile.
I need to make some code clean ups before the guys review the PR

File: src/chi/javascript/components/sidenav.js [hunk 2]
```diff
@@ -98,6 +95,41 @@ class Sidenav extends Component {
...
+          'click',
+          function(e) {
+            e.stopPropagation();
+          }
+        );
+        disableAnimation(drawer);
+      };
+
+      this._drawers.forEach((drawer) => {
+        drawer._elem.addEventListener('mouseenter', () => _openDrawerOnMouseEnter(drawer));
```
Comment (SergioQCL) [inline src/chi/javascript/components/sidenav.js:128]: Are you removing these events somewhere?

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/1144/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://c5d6a1524608790a8e06c3f929a776c1799a415d-chi-pr.enterprise.ctl.io" target="_blank"> https://c5d6a1524608790a8e06c3f929a776c1799a415d-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/1145/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://ddece7a8559f3bb3e9c6fc869dcc6f1a076960e2-chi-pr.enterprise.ctl.io" target="_blank"> https://ddece7a8559f3bb3e9c6fc869dcc6f1a076960e2-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1146/ for more details.

File: src/website/views/components/sidenav/_examples.pug [hunk 3]
```diff
@@ -483,6 +483,245 @@ h3 Inverse
 
       <script>chi.sidenav(document.getElementById('sidenav-1'));</script>
 
+h3 Open on hover
+- example +=1
+.example.-mb--3
+  .-d--flex(style='overflow: hidden;')
+    aside.chi-sidenav.-flex--shrink0#example-open-on-hover
```
Comment (mattnickles) [inline src/website/views/components/sidenav/_examples.pug:490]: This is rendering with the base theme but the code sample below has an `-inverse` class and renders in the inverse theme. Need to add the -inverse class here.

```suggestion
    aside.chi-sidenav.-inverse.-flex--shrink0#example-open-on-hover
```

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://d23b8dcef495a92498759f39be4313179583f9dd-chi-pr.enterprise.ctl.io" target="_blank"> https://d23b8dcef495a92498759f39be4313179583f9dd-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1147/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build FAILED. See http://jenkins.elasticbox.com/job/chi_pull_request/1149/ for more details.

**Comment by mattnickles** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://8ca7dfbddc339b275613e2345cdbd0d5504d7235-chi-pr.enterprise.ctl.io" target="_blank"> https://8ca7dfbddc339b275613e2345cdbd0d5504d7235-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1150/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-6ebswe, https://ops.elasticbox.com/services/instances/i-ws8pyr, https://ops.elasticbox.com/services/instances/i-mzmhfk, https://ops.elasticbox.com/services/instances/i-1xzyc4, https://ops.elasticbox.com/services/instances/i-lws4n4, https://ops.elasticbox.com/services/instances/i-p70qsr, https://ops.elasticbox.com/services/instances/i-3belw8, https://ops.elasticbox.com/services/instances/i-n9zt5s

**Pull request merged**